### PR TITLE
Checkout main branch in build-conda-pkg

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Github renamed the master branch of the feedstock to main so we need to update `build_conda_pkg`

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

**Review by jeremyliweishih — APPROVED** [review]
thanks!

File: .github/workflows/build_conda_pkg.yml [hunk 1]
```diff
@@ -6,10 +6,10 @@ on:
       - main
   pull_request:
     types: [ opened, synchronize ]
-    paths:
-      - 'requirements.txt'
-      - 'core-requirements.txt'
-      - 'evalml/tests/dependency_update_check/*.txt'
+#    paths:
```
Comment (fjlanasa) [inline .github/workflows/build_conda_pkg.yml:9]: Was this intentional?

Comment (freddyaboulton) [reply .github/workflows/build_conda_pkg.yml:9]: It’s just so that I could trigger the job. Will revert prior to merge!

**Review by chukarsten — APPROVED** [review]
LGTM when tests pass

**Pull request merged**